aws-flow/lib/aws/decider/task_poller.rb

```diff
@@ -191,6 +191,13 @@ def execute(task)
         end
       end
 
+      # Returns an array of activity names registered.
+      def activity_list
+        @activity_definition_map.keys.map do |key|
```

Comment:
This clearly assumes same class hierarchy for all the activities. Else we can implement something like longest common substring to pick the common classes and have activities defined like, 
ClassName1 (activity1.method,....)